docs/streams/developer-guide/dsl-api.html

```diff
@@ -3097,152 +3098,994 @@ <h5><a class="toc-backref" href="#id34">KTable-KTable Foreign-Key
+    }
+
+    private static class SlangReplacementTransformer implements ValueTransformer&lt;String, Iterable&lt;String&gt;&gt; {
+
+        @Override
+        public void init(final org.apache.kafka.streams.processor.ProcessorContext context) {
+        }
+
+        @Override
+        public Iterable&lt;String&gt; transform(final String value) {
+            if (value == null) {
+                return Collections.emptyList(); // Skip null values
+            }
+
+            // Replace slang words in the message
+            final String[] words = value.split(&quot;\\s+&quot;);
+            return Arrays.asList(
+                    Arrays.stream(words)
+                            .map(word -&gt; SLANG_DICTIONARY.getOrDefault(word, word))
+                            .toArray(String[]::new)
+            );
+        }
+
+        @Override
+        public void close() {
+        }
+    }
+
+    private static class SlangReplacementProcessor extends ContextualFixedKeyProcessor&lt;String, String, String&gt; {
+        @Override
+        public void process(final FixedKeyRecord&lt;String, String&gt; record) {
+            if (record.value() == null) {
+                return; // Skip null values
+            }
+
+            // Replace slang words in the message
+            final String[] words = record.value().split(&quot;\\s+&quot;);
+            for (final String word : words) {
+                String replacedWord = SLANG_DICTIONARY.getOrDefault(word, word);
+                context().forward(record.withValue(replacedWord));
+            }
+        }
+    }
+}</code></pre>
+                <h4>Keynotes</h4>
+                <ul>
+                    <li><strong>Type Safety and Flexibility:</strong> The process and processValues APIs utilize
+                        <code>ProcessorContext</code> and <code>Record</code> or<br><code>FixedKeyRecord</code> objects for better type
+                        safety and flexibility of custom processing logic.</li>
+                    <li><strong>Clear State and Logic Management:</strong> Implementations for <code>Processor</code> or
+                        <code>FixedKeyProcessor</code> should manage state and<br>logic clearly. Use <code>context().forward()</code>
+                        for emitting records downstream.</li>
+                    <li><strong>Unified API:</strong> Consolidates multiple methods into a single, versatile API.</li>
+                    <li><strong>Future-Proof:</strong> Ensures compatibility with the latest Kafka Streams releases.</li>
+                </ul>
+                <h4>Removal of Old <code>process</code> Method</h4>
+                <p>It is worth mentioning that, in addition to the methods mentioned above, the <code>process</code> method (which is
+                    not part of the<br>Processor API) has also been removed. The following example shows how to migrate to the new
```

Comment:
```
which is not part of the<br>Processor API
```
Not sure what you mean by this?

Maybe we want to day: "which did integrate the 'old' Processor API (ie,`Processor` in contrast to the new `api.Processor`) into the DSL", or something like this?

Reply:
Perfect! This is exactly what I wanted to say. Updated.